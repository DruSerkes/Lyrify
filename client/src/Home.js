import React, { useEffect, useMemo } from 'react';
import { useLocation } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { gotUser } from './reducers/actions';
import queryString from 'query-string';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';

const Home = () => {
	const dispatch = useDispatch();
	const user = useSelector((state) => state.user);
	const search = useLocation().search;
	const parsed = useMemo(() => queryString.parse(search), [ search ]);
	console.log('user === ', user);

	useEffect(
		() => {
			if (parsed.id) {
				console.log('parsed == ', parsed);
				localStorage.setItem('id', parsed.id);
				dispatch(gotUser(parsed));
			}
		},
		[ dispatch, parsed ]
	);

	return (
		<div>
			<Typography variant="h1">Lyrify</Typography>

			{user ? (
				<Typography variant="body1" margin={2}>
					Welcome back {user.display_name}
					{/* TODO Add copy */}
				</Typography>
			) : (
				<a href="http://localhost:5000/spotify/auth">
					<Button variant="contained" color="primary">
						{/* TODO Add copy */}
						Login with Spotify
					</Button>
				</a>
			)}
		</div>
	);
};

export default Home;
