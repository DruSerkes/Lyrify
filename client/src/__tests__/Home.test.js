import React from 'react';
import { render } from '@testing-library/react';
import Home from '../Home';
import { MemoryRouter } from 'react-router-dom';

it('renders without crashing', () => {
	render(
		<MemoryRouter>
			<Home />
		</MemoryRouter>
	);
});

it('matches snapshot', () => {
	const { asFragment } = render(
		<MemoryRouter>
			<Home />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});

it('should show login button to not-logged in user', () => {
	const { getByText } = render(
		<MemoryRouter>
			<Home />
		</MemoryRouter>
	);
	const btn = getByText('Login with Spotify');
	expect(btn).toBeInTheDocument();
});
