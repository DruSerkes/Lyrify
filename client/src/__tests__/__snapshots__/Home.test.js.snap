// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="MuiTypography-root MuiTypography-h1"
    >
      Lyrify
    </h1>
    <a
      href="http://localhost:5000/spotify/auth"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Login with Spotify
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </a>
  </div>
</DocumentFragment>
`;
